packages/flutter/lib/src/rendering/box.dart

```diff
@@ -1970,16 +1970,26 @@ abstract class RenderBox extends RenderObject {
         assert(size._owner == this);
         if (RenderObject.debugActiveLayout != null &&
             !RenderObject.debugActiveLayout!.debugDoingThisLayoutWithCallback) {
-          assert(
-            debugDoingThisResize || debugDoingThisLayout || _computingThisDryLayout ||
-              (RenderObject.debugActiveLayout == parent && size._canBeUsedByParent),
-            'RenderBox.size accessed beyond the scope of resize, layout, or '
-            'permitted parent access. RenderBox can always access its own size, '
-            'otherwise, the only object that is allowed to read RenderBox.size '
-            'is its parent, if they have said they will. It you hit this assert '
-            'trying to access a child\'s size, pass "parentUsesSize: true" to '
-            "that child's layout().",
-          );
+          // The Flutter pipeline is currently running a layout pass. 
+          // TODO: explain why we care that debugDoingThisLayoutWithCallback
+          
+          if (!debugDoingThisResize && !debugDoingThisLayout && !_computingThisDryLayout) {
+            // This RenderBox is not in the middle of running its own layout, which means some other
+            // object is trying to access the size.
+            if (RenderObject.debugActiveLayout != parent) {
+              throw AssertionError(
```

Comment:
Instead of an AssertionError can we be even more helpful with a full FlutterError? You can add an ErrorHint for these helpful suggestions to resolve the issue.